lib/jsdom/browser/Window.js

```diff
@@ -38,39 +34,62 @@ const WindowEventHandlersImpl = require("../living/nodes/WindowEventHandlers-imp
 // NB: the require() must be after assigning `module.exports` because this require() is circular
 // TODO: this above note might not even be true anymore... figure out the cycle and document it, or clean up.
 module.exports = Window;
-const dom = require("../living");
+const { installInterfaces } = require("../living/interfaces");
 
-dom.Window = Window;
+// https://html.spec.whatwg.org/#the-window-object
+function setupWindow(windowInstance) {
+  // Install first all the interfaces on the newly created window.
+  installInterfaces(windowInstance);
+
+  const EventTargetConstructor = windowInstance.EventTarget;
+
+  // eslint-disable-next-line func-name-matching, func-style, no-shadow
+  const windowConstructor = function Window() {
+    throw new TypeError("Illegal constructor");
+  };
+  Object.setPrototypeOf(windowConstructor, EventTargetConstructor);
+
+  Object.defineProperty(windowInstance, "Window", {
+    configurable: true,
+    writable: true,
+    value: windowConstructor
+  });
+
+  const windowPrototype = Object.create(EventTargetConstructor.prototype);
+  Object.defineProperties(windowPrototype, {
+    constructor: {
+      value: windowConstructor,
+      writable: true,
+      configurable: true
+    },
+    [Symbol.toStringTag]: {
+      value: "Window",
+      configurable: true
+    }
+  });
+
+  windowConstructor.prototype = windowPrototype;
+  Object.setPrototypeOf(windowInstance, windowPrototype);
+
+  EventTarget.setup(windowInstance, windowInstance);
+  mixin(windowInstance, WindowEventHandlersImpl.prototype);
+  mixin(windowInstance, GlobalEventHandlersImpl.prototype);
+  windowInstance._initGlobalEvents();
+
+  windowInstance._globalObject = windowInstance;
+}
 
 // NOTE: per https://heycam.github.io/webidl/#Global, all properties on the Window object must be own-properties.
 // That is why we assign everything inside of the constructor, instead of using a shared prototype.
 // You can verify this in e.g. Firefox or Internet Explorer, which do a good job with Web IDL compliance.
-
 function Window(options) {
```

Comment:
Hmm, the weirdness where this is named `Window` but the actual `Window` constructor is created on line 47 is strange... I'm not sure what the right architecture is. I guess we can try to straighten it out later, or just wait until we move Window to webidl2js.

Reply:
I wanted to minimize the amount of code change made to `lib/jsdom/browser/Window.js`, a clean up is overdue. Adding support for `[Global]` and `[NamedConstructor]` would greatly help. Let's do that in another PR.